src/main/java/com/google/sps/scheduler/Scheduler.java

```diff
@@ -39,38 +41,53 @@
   public static final int END_HOUR = 18;
   public static final int END_MINUTE = 0;
   public static final long DEFAULT_DURATION_IN_MILLISECONDS = TimeUnit.MINUTES.toMillis(30);
+  private String timeZone;
+  private TreeMultimap<Long, ExtendedTask> longestFirstOrderedTasks;
+  private List<ExtendedTask> tasks;
+  private Set<Event> orderedCalendarEvents;
+  private List<ExtendedTask> scheduledTasks;
+
+  public Scheduler(Collection<Event> calendarEvents, List<ExtendedTask> tasks, String timeZone) {
+    this.timeZone = timeZone;
+    this.tasks = tasks;
+
+    orderedCalendarEvents = new TreeSet<>(
+        Comparator.comparingLong(event -> event.getStart().getDateTime().getValue()));
+    orderedCalendarEvents.addAll(calendarEvents);
+
+    // It is possible to omit the parameters, but then we would have to implement a comparator on the tasks
+    longestFirstOrderedTasks = TreeMultimap.create(Ordering.natural(), Ordering.arbitrary());
+    for (ExtendedTask task : this.tasks) {
+      longestFirstOrderedTasks.put(task.getDuration(), task);
+    }
+  }
 
   /**
    * Schedules the tasks in the free time slot of the calendar events, which must be of the same day
    * specified in the parameter.
    * For each task that can be scheduled, the due time is set and a list of all scheduled tasks
    * is returned.
    */
-  public static List<Task> scheduleInRange(List<Event> calendarEvents, List<Task> tasks, String timeZone, LocalDate startDate, LocalDate endDate) {
+  public List<ExtendedTask> scheduleInRange(LocalDate startDate, LocalDate endDate) {
```

Comment:
This function should create a copy of longestFirstOrderedTasks (or initialize longestFirstOrderedTasks here, and not in the constructor), otherwise a scheduler can only do `scheduleInRange` once. Or this should be clearly indicated in the class description.

Reply:
I think it is useful, if the longestFirstOrderedTasks is a field of the class, so that each method can easily access the same data structure. I would go with the "adding class description" solution.

Reply:
I mean this is obviously usefull. It is the matter of where to initialize it.

Reply:
Especially since you had local a list of original tasks in the constructor. So it was just a matter of moving creation of the treemap to the function

Reply:
Okay.

Reply:
(Just a quick note, I think usually we do not expect to schedule the same tasks ever again.)